app/api/api_v1/routers/collection.py

```diff
@@ -69,27 +75,50 @@ async def get_all_collections() -> list[CollectionReadDTO]:
     "/collections/",
     response_model=list[CollectionReadDTO],
 )
-async def search_collection(q: str = "") -> list[CollectionReadDTO]:
+async def search_collection(request: Request) -> list[CollectionReadDTO]:
     """
-    Searches for collections matching the "q" URL parameter.
-
-    :param str q: The string to match, defaults to ""
-    :raises HTTPException: If nothing found a 404 is returned.
-    :return list[CollectionDTO]: A list of matching collections.
+    Searches for collections matching URL parameters ("q" by default).
+
+    :param Request request: The fields to match against and the values
+        to search for. Defaults to searching for "" in collection titles
+        and summaries.
+    :raises HTTPException: If invalid fields passed a 400 is returned.
+    :raises HTTPException: If a DB error occurs a 503 is returned.
+    :raises HTTPException: If the search request times out a 408 is
+        returned.
+    :return list[CollectionReadDTO]: A list of matching collections
+        (which can be empty).
     """
+
+    query_params = get_query_params_as_dict(request.query_params)
+
+    DEFAULT_SEARCH_FIELDS = ["title", "description"]
```

Comment:
I wonder if you could explain `DEFAULT_SEARCH_FIELDS` for all the entities. It appears as though this is used to remove duplicates from the query_params - but it seems in the repository for the entities these fields are hard coded - so not sure why we use them?

Reply:
Very good point. Updated.